fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -24,115 +23,127 @@ class FulladlePlugin : Plugin<Project> {
-            configureLibraryModule(this, flankGradleExtension)
+          if (!hasAndroidTest)
+            return@subprojects
+          allModulesDisabled = false
+          if (isAndroidAppModule)
+            configureModule(this, flankGradleExtension)
+        }
+        // then configure all library modules
+        root.subprojects {
+          if (!hasAndroidTest)
+            return@subprojects
+          allModulesDisabled = false
+          if (isAndroidLibraryModule)
+            configureModule(this, flankGradleExtension)
+        }
+        check(!allModulesDisabled) {
+          "All modules were disabled for testing in fulladleModuleConfig or the enabled modules had no tests"
         }
       }
     }
 
+    root.tasks.withType(YamlConfigWriterTask::class.java).configureEach {
+      dependsOn(fulladleConfigureTask)
+    }
+
     root.afterEvaluate {
       // TODO add other printYml tasks from other configs
       root.tasks.named("printYml").configure {
         dependsOn(fulladleConfigureTask)
       }
-
-      root.tasks.withType(YamlConfigWriterTask::class.java).configureEach {
-        dependsOn(fulladleConfigureTask)
-      }
     }
   }
 }
 
-fun configureLibraryModule(project: Project, flankGradleExtension: FlankGradleExtension) = project.run {
-  pluginManager.withPlugin("com.android.library") {
-    val fulladleModuleExtension = extensions.getByType(FulladleModuleExtension::class.java)
-    if (!fulladleModuleExtension.enabled.get())
-      return@withPlugin
-
-    val library = extensions.getByType<LibraryExtension>()
-
-    library.testVariants.configureEach {
-      if (!file("$projectDir/src/androidTest").exists()) {
-        println("Ignoring $name test variant in $path: No tests in $projectDir/src/androidTest")
-        return@configureEach
-      }
-
-      val debugApk = propertyToYaml(fulladleModuleExtension.debugApk, "app")
-
-      val maxTestShards = propertyToYaml(fulladleModuleExtension.maxTestShards, "max-test-shards")
-
-      val clientDetails = mapPropertyToYaml(fulladleModuleExtension.clientDetails, "client-details")
+fun configureModule(project: Project, flankGradleExtension: FlankGradleExtension) = project.run {
+  val fulladleModuleExtension = extensions.findByType(FulladleModuleExtension::class.java) ?: return
+  if (!hasAndroidTest) {
+    return
+  }
 
-      val environmentVariables = mapPropertyToYaml(fulladleModuleExtension.environmentVariables, "environment-variables")
+  val testedExtension = extensions.getByType<TestedExtension>()
+  // Only configure the first test variant per module.
+  // Does anyone test more than one variant per module?
+  var addedTestsForModule = false
 
-      outputs.configureEach {
-        val strs = listOf("- test: $outputFile", debugApk, maxTestShards, clientDetails, environmentVariables)
-        writeAdditionalTestApps(strs, flankGradleExtension, rootProject)
-      }
+  // TODO deal with ignored/filtered variants
```

Comment:
:( i don't like TODOs in code, but it isnt part of this PR. we should explain the limitations instead of having a TODO.